src/main.rs

```diff
@@ -172,3 +177,32 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    env_logger::init();
+
+    // Show a splash screen very early on
+    let display = setup_display();
+
+    match init().await {
+        Ok((ui, http_server, watchdog)) => run(ui, http_server, watchdog, display).await,
+        Err(e) => {
+            // Display a detailed error message on stderr (and thus in the journal) ...
+            error!("Failed to initialize tacd: {e}");
+
+            // ... and a generic message on the LCD, as it can not fit a lot of detail.
+            display.clear();
+            display.with_lock(|target| {
+                message(
+                    target,
+                    "tacd failed to start.\nCheck log for info.\nWaiting for watchdog.",
```

Comment:
```suggestion
                    "tacd failed to start.\nCheck log for info.\nWaiting for watchdog,\n to reset the system.",
```

Reply:
Perhaps a little explanation what the watchdog will do is a good for UX?

Reply:
Sounds good. The watchdog in question is however only a systemd software watchdog (which in turn is however supervised by the hardware watchdog). Systemd will try to restart the tacd. I've updated the message accordingly.

One of the reasons to keep the number of lines low was that I did not actually center the message vertically as I thought that would involve counting lines and calculations based on the line height.
That was not the case. I've now added actual centering which was actually quite easy to do.

Reply:
Great and thanks for the insight.

> I've now added actual centering which was actually quite easy to do.

Yes the embedded graphics crate has some nice features in that regard.